mempool/clist_mempool_test.go

```diff
@@ -292,14 +292,33 @@ func TestMempoolUpdate(t *testing.T) {
 	}
 }
 
+func TestMempoolLanesValidation(t *testing.T) {
```

Comment:
I think a test scenario with empty `lanePriorities` and `defLane` = 0 is missing.

Reply:
That should be the test at L298 because when you initialize `appInfo` the list is empty and the value for `DefaultLanePriority` is 0. 

Note that in line 296 `appInfo` is just an initialization of the `InfoResponse` struct with default values. It is not an actual call to the application.

Reply:
Do you think it is necessary to define a default line? What if the application sends you a list of lanes without 1?

Reply:
The ADR mentions the validation rule "the list lane_priorities is empty if and only if default_lane_priority is 0." but I think we can relax this rule a bit by allowing the app to have a non-empty list of lanes without a default lane. Then on a CheckTxResponse that doesn't specify a lane, the mempool will just assign the default lane `types.Lane(1)`, which must exists always on the mempool side (as defined in the ADR).

Reply:
I agree, but what if the non empty list of lanes does not contain our predefined default lane (1)? From what I understood the list of priorities does not have to be a contiguous array. 

So a user provides you a list of priorities :{2,4,6} and we set the default priority to {1}. This will then throw an error that the default priority is not in the priority list.

Reply:
@hvanz , @sergio-mena I still think that having a non empty list of priorities without explicitly setting a default priority (defaultPriority is 0), should be an error. If you feel strongly against that, I can remove this as an error case.